### PR TITLE
Add support for s390x architecture.

### PR DESCRIPTION
**Pull request by PiotrSikora**
### Description of changes: 

Add support for s390x architecture.

This is pretty trivial change (a few defines), since the big-endian support was previously added for PowerPC architectures.

### Call-outs:

There is a false-positive error when compiling using `s390x-linux-gnu-gcc-11` with `-O3` (i.e. `Release` profile):
```
crypto/pem/pem_lib.c:705:11: error: ‘strncmp’ of strings of length 1 and 9 and bound of 9 evaluates to nonzero [-Werror=string-compare]
 705 |       if (strncmp(buf, "-----END ", 9) == 0) {
     |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```
but there are no issues when using `-O2` or when building with `s390x-linux-gnu-gcc-12`.

### Testing:

All tests from `run_cross_tests.sh`, i.e.
```
crypto/crypto_test --gtest_also_run_disabled_tests
crypto/urandom_test
crypto/mem_test
crypto/mem_set_test
crypto/dynamic_loading_test

ssl/ssl_test
ssl/integration_test
```
passed, with the exception of `BIOTest.InvokeConnectCallback`, which fails for me on all platforms.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

cc @andrewhop @justsmth

### CODE REVIEW

**Comment by justsmth** [comment]
Could this be added to our [cross-tests](https://github.com/aws/aws-lc/blob/main/.github/workflows/cross-test.yml)?

**Comment by PiotrSikora** [comment]
> Could this be added to our [cross-tests](https://github.com/aws/aws-lc/blob/main/.github/workflows/cross-test.yml)?

It can be added, but I believe that someone from your team has to upload the cross-compiling toolchain for s390x, which is downloaded as part of the CI (see: [tests/ci/run_cross_tests.sh#L34](https://github.com/aws/aws-lc/blob/56def5a253d27280f3b7bd6564cfa5a11211aee8/tests/ci/run_cross_tests.sh#L34)).

> (After fixing `BIOTest.InvokeConnectCallback`?)

As mentioned in the original message, this test is always failing for me without any code changes in all the configurations I've tested (i.e. when running unmodified `main` branch on native x86-64 system, and also when cross-compiling to different architectures), and I don't believe there is anything s390x-specific about this failure.

Anyway, it looks that the test was broken: https://github.com/aws/aws-lc/pull/1433.

File: crypto/fipsmodule/rand/getrandom_fillin.h [hunk 1]
```diff
@@ -34,6 +34,8 @@
 #define EXPECTED_NR_getrandom 359
 #elif defined(OPENSSL_RISCV64)
 #define EXPECTED_NR_getrandom 278
+#elif defined(OPENSSL_S390X)
+#define EXPECTED_NR_getrandom 349
```
Comment (justsmth) [inline crypto/fipsmodule/rand/getrandom_fillin.h:38]: For future reference, this value aligns [with `getrandom` in the the syscall table](https://github.com/torvalds/linux/blob/99bd3cb0d12e85d5114425353552121ec8f93adc/arch/s390/kernel/syscalls/syscall.tbl#L359).

**Comment by justsmth** [comment]
Sorry for the delay in reviewing this. I plan to build a toolchain to perform testing for this platform. I'll hopefully have an update on it soon.

**Comment by justsmth** [comment]
CI tests for the s390x passed: https://github.com/aws/aws-lc/actions/runs/8116237276/job/22185808912?pr=1431

**Review by justsmth — APPROVED** [review]

**Review by nebeid — APPROVED** [review]

**Pull request merged**